File: client/android/UseCase/Auth/Session/ChallengeSecondFactorUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.data.auth.session.AuthSessionGateway

class ChallengeSecondFactorUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(): AuthSessionPayload = gateway.challengeSecondFactor()
}

File: client/ios/UseCase/Auth/Session/ChallengeSecondFactorUseCase.swift
public struct ChallengeSecondFactorUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction() async throws -> AuthSessionPayload {
        try await gateway.challengeSecondFactor()
    }
}
